### PR TITLE
React 16 compat & Remove sticky popover

### PR DESCRIPTION
**Pull request by kristofferj**
NOT READY TO MERGE YET

Problem: Forms inside block-editor popup looses focus.

### CODE REVIEW

**Comment by bjoerge** [comment]
Rebased against current next

File: packages/@sanity/components/src/selects/StatelessSearchableSelect.js [hunk 1]
```diff
@@ -170,85 +174,88 @@ export default class StatelessSearchableSelect extends React.PureComponent {
...
-                          onSelect={this.handleSelect}
-                          renderItem={renderItem}
-                          highlightIndex={highlightIndex}
-                        />
-                      )}
+                      }
+                    }
+                  >
+                    <div>
+                      <CaptureOutsideClicks onClickOutside={(isActive && isOpen) ? this.handleClose : undefined}>
```
Comment (bjoerge) [inline packages/@sanity/components/src/selects/StatelessSearchableSelect.js]: For consistency with other usages of this pattern: Use `(isActive && isOpen) ? this.handleClose : null` (`null` instead of `undefined`) here too

File: packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js [hunk 2]
```diff
@@ -389,13 +386,13 @@ export default class FormBuilderBlock extends React.Component {
         onDragLeave={this.handleCancelEvent}
         onDrop={this.handleCancelEvent}
         draggable
-        onClick={this.handleToggleEdit}
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js]: Why was this moved? This is the cause of:
> Problem: Forms inside block-editor popup loses focus.

Comment (bjoerge) [reply packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js]: Ehhh, nevermind that, sorry.

File: packages/@sanity/components/src/dialogs/PopOver.js [hunk 3]
```diff
@@ -1,152 +1,129 @@
...
+import {Portal} from 'react-portal'
 import Stacked from '../utilities/Stacked'
 import CaptureOutsideClicks from '../utilities/CaptureOutsideClicks'
 import Escapable from '../utilities/Escapable'
+import {Manager, Target, Popper, Arrow} from 'react-popper'
 
-const PADDING = 20
-
-export default class PopOver extends React.Component {
+export default class EditItemPopOver extends React.Component {
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/dialogs/PopOver.js]: yep

File: packages/@sanity/components/src/selects/StatelessSearchableSelect.js [hunk 4]
```diff
@@ -7,14 +7,15 @@ import DefaultTextInput from 'part:@sanity/components/textinputs/default'
...
 import Stacked from '../utilities/Stacked'
 import CaptureOutsideClicks from '../utilities/CaptureOutsideClicks'
 import Escapable from '../utilities/Escapable'
+import {Portal} from 'react-portal'
+import {Manager, Target, Popper} from 'react-popper'
 
 const noop = () => {}
 
-export default class StatelessSearchableSelect extends React.PureComponent {
+export default class StatelessSearchableSelect extends React.Component {
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/selects/StatelessSearchableSelect.js]: This was unintentional. (i wasprobably just testing something) Have reverted

File: packages/@sanity/components/src/selects/SearchableSelect.js [hunk 5]
```diff
@@ -166,7 +166,6 @@ export default class SearchableSelect extends React.Component {
           isInputSelected={isInputSelected}
           inputValue={inputValue}
           onInputChange={this.handleInputChange}
-          width={width}
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/selects/SearchableSelect.js]: Yes. I now also removed every resize-handler and make popper handle this.

**Comment by kristofferj** [comment]
The popover now seems to work by changing the order of portal/stacked etc. However, the stacked components seems to be confused over its active state. Needs investigation. Seems to affect both popover, fullscreendialog and defaultdialog.

File: packages/@sanity/components/src/buttons/createButtonLike.js [hunk 6]
```diff
@@ -34,14 +34,6 @@ export default function createButtonLike(Component, {displayName, defaultProps =
       ...defaultProps
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/buttons/createButtonLike.js]: Seems there are no changes anymore?

File: packages/@sanity/base/src/styles/layout/scrolling.css [hunk 7]
```diff
@@ -1,22 +1,14 @@
 @import 'part:@sanity/base/theme/variables-style';
 
 :root {
-  --defaultScrollBarWidth: var(--small-padding);
+  --defaultScrollBarWidth: calc(var(--small-padding) / 2);
 }
 
 .lightScrollbars {
-  @media (--screen-medium) {
+  /* @media (--screen-medium) {
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/base/src/styles/layout/scrolling.css]: I decided to keep it. Should do this in a seperate branch if we decide to do.

File: packages/@sanity/components/src/selects/StatelessSearchableSelect.js [hunk 8]
```diff
@@ -115,12 +110,11 @@ export default class StatelessSearchableSelect extends React.PureComponent {
...
   }
 
-  setListElement = element => {
-    this._listElement = element
-  }
-
-  setRootNode = node => {
-    this._rootNode = node
+  handleClose = event => {
+    if (this.props.isOpen) {
```
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/selects/StatelessSearchableSelect.js]: Seems so. Did remove.

File: packages/@sanity/components/src/buttons/createButtonLike.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (bjoerge) [reply packages/@sanity/components/src/buttons/createButtonLike.js]: Still changes in this file: https://github.com/sanity-io/sanity/pull/639/files#diff-f8ca333bd9b58b674f3880ba76e93d42L61

File: packages/@sanity/components/src/selects/StatelessSearchableSelect.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (bjoerge) [reply packages/@sanity/components/src/selects/StatelessSearchableSelect.js]: Looks like the change is still there?

File: packages/@sanity/components/src/buttons/createButtonLike.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (kristofferj) [reply packages/@sanity/components/src/buttons/createButtonLike.js]: Ok. Did some digging. This needs to be included, cause popperjs dont detect overflow: overlay;. Turns out that overlay is deprecated, that need to go. This means that when hiding scrollbars, and hover to show them resizes the content. Changes in this file makes the scrollbar thinner and does not resize the content on overflow. Needs an improvement in the future, in another branch, but these changes are connected to removing the sticky popover and introducing popperjs.

[the remark this answers is not in the document]
Comment (bjoerge) [reply packages/@sanity/components/src/buttons/createButtonLike.js]: Got it, thanks!
